airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -91,17 +69,15 @@ def chunk_date_range(
 
     # applying conversion window
     start_date = start_date.subtract(days=conversion_window)
+    slice_start = start_date
 
-    while start_date < end_date:
-        start, end = get_date_params(start_date.to_date_string(), time_zone=time_zone, range_days=range_days, end_date=end_date)
-        intervals.append(
-            {
-                "start_date": start,
-                "end_date": end,
-            }
-        )
-        start_date = start_date.add(days=range_days)
-    return intervals
+    while slice_start.date() <= end_date.date():
+        slice_end = min(end_date, slice_start.add(days=range_days - 1))
+        yield {
+            "start_date": slice_start.to_date_string(),
```

Comment:
So just to confirm the API behavior, Google's API fetches all records inclusive of the start date passed in and inclusive of the end date passed in? And in the prior code, we took the start date and added 1 day. Thus how we ended up with the `>` operation on the start date?

Reply:
Yes, both start date and end date are inclusive. We did not actually end up with `>`, the problem was with the start date of the initial slice. If the user used to enter `2020-01-01` as a start date, we would generate slices starting from `2020-01-02`, that's the whole point. Now having refactored the code, slicing starts with `2020-01-01` as expected

Reply:
perfect, thanks for confirming!